### PR TITLE
expand: replace getopts with clap

### LINKED ISSUE #1795: expand: move from getopts to clap
Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/expand/src/expand.rs

Tests:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_expand.rs (need to add tests)

As example:
https://github.com/uutils/coreutils/commit/9081e120d6b866c94679d9e29c7bdb17ccccb4d0

### PR DESCRIPTION
**Pull request by deantvv**
expand has one odd behavior that allows two format for tabstop

From expand --help
```
-t, --tabs=N     have tabs N characters apart, not 8
-t, --tabs=LIST  use comma separated list of tab positions
```

This patch use one `value_name("N, LIST")` for tabstop and
deal with above behavior in `parse_tabstop`.

Close #1795

### CODE REVIEW

File: src/uu/expand/src/expand.rs [hunk 1]
```diff
@@ -58,14 +64,14 @@ struct Options {
...
+    fn new(matches: &ArgMatches) -> Options {
+        let tabstops = match matches.value_of("tabstops") {
+            Some(s) => tabstops_parse(s.to_string()),
             None => vec![DEFAULT_TABSTOP],
-            Some(s) => tabstops_parse(s),
         };
 
-        let iflag = matches.opt_present("i");
-        let uflag = !matches.opt_present("U");
+        let iflag = matches.is_present("iflag");
```
Comment (sylvestre) [inline src/uu/expand/src/expand.rs:73]: Please follow the same declarations as in other for the options :)
For example:
https://github.com/uutils/coreutils/blob/517b2a6a2244637a5f5331565592873a734085c7/src/uu/chown/src/chown.rs#L30

File: src/uu/expand/src/expand.rs [hunk 2]
```diff
@@ -80,10 +86,9 @@ impl Options {
             .unwrap(); // length of tabstops is guaranteed >= 1
         let tspaces = repeat(' ').take(nspaces).collect();
 
-        let files = if matches.free.is_empty() {
-            vec!["-".to_owned()]
-        } else {
-            matches.free
+        let files: Vec<String> = match matches.values_of("files") {
+            Some(s) => s.map(|v| v.to_string()).collect(),
+            None => vec!["-".to_owned()],
```
Comment (sylvestre) [inline src/uu/expand/src/expand.rs:91]: Would be great to have a test covering this case :)

File: src/uu/expand/src/expand.rs [hunk 3]
```diff
@@ -97,31 +102,34 @@ impl Options {
...
-            "have tabs NUMBER characters apart, not 8",
-            "NUMBER",
+    let usage = get_usage();
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .after_help(LONG_HELP)
+        .arg(
+            Arg::with_name("iflag")
```
Comment (sylvestre) [inline src/uu/expand/src/expand.rs:112]: ditto for the options :)

File: src/uu/expand/src/expand.rs [hunk 4]
```diff
@@ -97,31 +102,34 @@ impl Options {
...
+                .value_name("N, LIST")
+                .takes_value(true)
+                .help("have tabs N characters apart, not 8 or use comma separated list of explicit tab positions"),
         )
-        .optflag(
-            "U",
-            "no-utf8",
-            "interpret input file as 8-bit ASCII rather than UTF-8",
+        .arg(
+            Arg::with_name("uflags").long("no-utf8").short("U").help("interpret input file as 8-bit ASCII rather than UTF-8"),
```
Comment (sylvestre) [inline src/uu/expand/src/expand.rs:126]: please split in multiple line (just like tabstops) for consitency

**Review by sylvestre — CHANGES_REQUESTED** [review]
Looks great, just some minor fixes!

File: src/uu/expand/src/expand.rs (hunk 3 above)
Comment (deantvv) [reply src/uu/expand/src/expand.rs]: Fix in cd775ed

File: src/uu/expand/src/expand.rs (hunk 2 above)
Comment (deantvv) [reply src/uu/expand/src/expand.rs:98]: Add test in f608084

File: src/uu/expand/src/expand.rs (hunk 1 above)
Comment (deantvv) [reply src/uu/expand/src/expand.rs]: Thanks for the advice. cd775ed fix this.

**Comment by deantvv** [comment]
@sylvestre Is there anything I could change to make this better?

(MinRustV failing is probably not because of anything in this pull request.)

**Comment by sylvestre** [comment]
Excellent, thanks!

**Pull request merged**